sfepy/terms/terms_volume.py

```diff
@@ -26,3 +26,54 @@ def get_fargs(self, mat, virtual,
         vg, _ = self.get_mapping(virtual)
 
         return mat, vg
+
+
+class NonLinearVolumeForceTerm(Term):
+    """
+     Product of virtual and function of state.
```

Comment:
The volume force term with the force given by a user supplied function of the state variable.